### PR TITLE
fix: Added region and signature for S3 bucket

### PR DESCRIPTION
**Pull request by farhaanbukhsh**
For a region based bucket boto3 needs to know about region and signature which edx-ora2 doesn't know about.

**TL;DR -** Currently edx-ora2 doesn't pick up AWS region or aws signature version and because of that boto3 generates a faulty S3 URL. The [boto3 documentation for s3 signature v4](https://boto3.amazonaws.com/v1/documentation/api/1.9.42/guide/s3.html#generating-presigned-urls) also shows the boto3 client need to know the signature version.

JIRA: None

**What changed?**

- Config to add AWS configuration for S3

When I was trying to upload the file in S3 bucket in the me-south-1 region, I kept on getting 
image

The reason was boto3 client doesn't use S3_SIGNATURE=`s3v4` and region value here in edx-ora2 while the same config was working properly on discussions.

**Developer Checklist**

- ✅ Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- ❌  Translations and JS/SASS compiled
- ❌  Bumped version number in [setup.py](https://github.com/openedx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/openedx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

- None since it just adds configs

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @farhaanbukhsh!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Review by kaustavb12 — APPROVED** [review]
:+1:

- [X] I tested this: Tested upload file to S3 works in sandbox
- [X] I read through the code
- [ ] I checked for accessibility issues
- [ ] Includes documentation
- [ ] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.

**Comment by itsjeyd** [comment]
Thanks @farhaanbukhsh and @kaustavb12!

Looks like this PR is ready for engineering review. I changed its status on the contributions board accordingly.

**Comment by itsjeyd** [comment]
Hi @mattcarter, if you could help get this PR lined up for engineering review by the Aurora team, that would be great!

**Comment by farhaanbukhsh** [comment]
@mattcarter Gentle ping on the PR.

**Comment by itsjeyd** [comment]
label: core contributor

**Comment by itsjeyd** [comment]
Hi @mattcarter, just checking in to see if you had a chance to line this PR up for engineering review by Aurora?

File: openassessment/fileupload/backends/s3.py [hunk 1]
```diff
@@ -75,12 +77,18 @@ def _connect_to_s3():
     aws_access_key_id = getattr(settings, "AWS_ACCESS_KEY_ID", None)
     aws_secret_access_key = getattr(settings, "AWS_SECRET_ACCESS_KEY", None)
     endpoint_url = getattr(settings, "AWS_S3_ENDPOINT_URL", None)
+    signature_version = getattr(settings, "AWS_S3_SIGNATURE_VERSION ", None)
```
Comment (Agrendalath) [inline openassessment/fileupload/backends/s3.py:80]:
```suggestion
    signature_version = getattr(settings, "AWS_S3_SIGNATURE_VERSION", None)
```

**Comment by Agrendalath** [comment]
@farhaanbukhsh, we may also want to bump the version in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3).

**Comment by farhaanbukhsh** [comment]
> @farhaanbukhsh, we may also want to bump the version in [openassessment/__init__.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3).

Thank you for the review @Agrendalath will make the changes :)

**Comment by itsjeyd** [comment]
@farhaanbukhsh It looks like we'll be able to proceed with a CC review here soon ([Slack ref](https://openedx.slack.com/archives/C03R320AFJP/p1687856711143529?thread_ts=1687251245.596839&cid=C03R320AFJP)). Could you please address @Agrendalath's comments in the coming days to get ready for that?

**Comment by farhaanbukhsh** [comment]
@itsjeyd @Agrendalath  Thank you for the feedback, I have made the changes and pushed the commit, :) please have a look.

**Review by Agrendalath — APPROVED** [review]
The changes look good to me 👍🏻

File: openassessment/fileupload/backends/s3.py [hunk 2]
```diff
@@ -8,6 +8,8 @@
 import botocore
 import boto3
 
+from botocore.client import Config
```
Comment (pomegranited) [inline openassessment/fileupload/backends/s3.py:11]: nit: any reason not to use the already-imported botocore library above?
Then below, it would just be

```
config=botocore.client.Config(
...
```

File: openassessment/fileupload/backends/s3.py [hunk 3]
```diff
@@ -75,12 +77,18 @@ def _connect_to_s3():
...
+    region_name = getattr(settings, "AWS_S3_REGION_NAME", None)
 
     return boto3.client(
         "s3",
         aws_access_key_id=aws_access_key_id,
         aws_secret_access_key=aws_secret_access_key,
         endpoint_url=endpoint_url,
+        config=Config(
+            signature_version=signature_version,
+            region_name=region_name
```
Comment (pomegranited) [inline openassessment/fileupload/backends/s3.py:90]: nit:
```suggestion
            region_name=region_name,
```

File: openassessment/__init__.py [hunk 4]
```diff
@@ -1,4 +1,4 @@
 """
 Initialization Information for Open Assessment Module
 """
-__version__ = '5.0.4'
+__version__ = '5.0.5'
```
Comment (pomegranited) [inline openassessment/__init__.py:4]: nit: [`master`](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) is currently at `'5.1.0'`, so:
```suggestion
__version__ = '5.1.1'
```

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "edx-ora2",
-  "version": "5.0.4",
+  "version": "5.0.5",
```
Comment (pomegranited) [inline package.json:3]: [ditto](https://github.com/openedx/edx-ora2/pull/1928/files#diff-60d133f8459119b2c44feadd371cc76e9d88161518789eefc8173b2d60197cb6R4):
```suggestion
  "version": "5.1.1",
```

**Review by pomegranited — CHANGES_REQUESTED** [review]
Thanks for this fix @farhaanbukhsh !

I left a few minor suggestions, but I've tested this in both scenarios below, and it works fine. So I'll approve when the nits are addressed.

* uploading to bucket in `us-east-1` with defaults left at `None` 
* uploading to bucket in `us-east-2` with `signature_version = "s3v4"`, `region_name = "us-east-2"`

File: openassessment/fileupload/backends/s3.py (hunk 2 above)
Comment (farhaanbukhsh) [reply openassessment/fileupload/backends/s3.py:11]: It was just looking cleaner :) and there was just one occurrence, I have corrected that now. :)

**Comment by farhaanbukhsh** [comment]
@pomegranited I have updated the code with your suggestions. Thanks a lot for the review :)

**Review by pomegranited — APPROVED** [review]
:+1: perfect, thank you @farhaanbukhsh !

Since I've [got permission to merge](https://github.com/openedx/edx-ora2/pull/1928#issuecomment-1621239492), I'll do that now, and tag a new release.

- [x] I tested this on my devstack:
   * left defaults at None, and tested file uploads to a bucket in `us-east-1`.
   * set signature_version = "s3v4", region_name = "us-east-2" and tested file uploads to a bucket in that region.
- [x] I read through the code
- [x] <strike>I checked for accessibility issues</strike> N/A
- [x] <strike>Includes documentation</strike> N/A

**Comment by openedx-webhooks** [comment]
@farhaanbukhsh 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-ora2&entry.867055334=https://github.com/openedx/edx-ora2/pull/1928&entry.1484655318=https://github.com/farhaanbukhsh&entry.752974735=2023-03-24+07:13&entry.1917517419=2023-07-12+07:37&entry.2133058324=Yes) so we can improve your experience in the future.
